python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -363,19 +363,16 @@ def run(self, calexpRefList, skyInfo, visitId=0):
         for calExpInd, calExpRef in enumerate(calexpRefList):
             self.log.info("Processing calexp %d of %d for this Warp: id=%s",
                           calExpInd+1, len(calexpRefList), calExpRef.dataId)
-            try:
-                ccdId = calExpRef.get("ccdExposureId", immediate=True)
-            except Exception:
-                ccdId = calExpInd
+            ccdId = calExpRef.get("ccdExposureId", immediate=True)
             try:
                 # We augment the dataRef here with the tract, which is harmless for loading things
-                # like calexps that don't need the tract, and necessary for meas_mosaic outputs,
+                # like calexps that don't need the tract, and necessary for jointcal outputs,
                 # which do.
                 calExpRef = calExpRef.butlerSubset.butler.dataRef("calexp", dataId=calExpRef.dataId,
                                                                   tract=skyInfo.tractInfo.getId())
                 calExp = self.getCalibratedExposure(calExpRef, bgSubtracted=self.config.bgSubtracted)
-            except Exception as e:
-                self.log.warn("Calexp %s not found; skipping it: %s", calExpRef.dataId, e)
+            except MissingExposureError as e:
+                self.log.warn("Skipping missing data: %s", e)
```

Comment:
Why no longer include dataId? I would imagine the dataId of the missing calexp useful for debugging, though such information may be obtained through other means.

Reply:
The assumption (and implementation) here is that the raised exception contains all the necessary information. Previously, the dataId was being duplicated.

Reply:
The only error from `getCalibratedExposure` that will include the `dataId` are failed butler calls.  Anything else goes wrong and we won't have the dataId logged. 

 Note: I'm moving all the Gen2 dataRef's out of `run` in the next month anyway.